tfutils/base.py

```diff
@@ -593,17 +719,63 @@ def run_base(saver_params,
                   'thres_loss': thres_loss,
                   'num_steps': num_steps,
                   'log_device_placement': log_device_placement}
-        for sk in ['host', 'port', 'dbname', 'collname', 'exp_id']:
-            assert sk in saver_params, (sk, saver_params)
-        saver = Saver(sess=sess, global_step=global_step, params=params, **saver_params)
-        run(sess,
-            queues,
-            saver,
-            train_targets=train_targets,
-            global_step=global_step,
-            num_steps=num_steps,
-            valid_targets=valid_targets_dict,
-            thres_loss=thres_loss)
+        dbinterface = DBInterface(sess=sess, global_step=global_step, params=params,
+                                  save_params=save_params, load_params=load_params)
+        dbinterface.initialize()
+        train(sess,
+              queues,
+              dbinterface,
+              train_targets=train_targets,
+              global_step=global_step,
+              num_steps=num_steps,
+              valid_targets=valid_targets_dict,
+              thres_loss=thres_loss)
+
+
+def get_valid_targets_dict(validation_params, 
+                           model_func, model_kwargs,
+                           default_queue_params,
+                           cfg_final):
+    """NB: this function may modify validation_params"""
+    valid_targets_dict = OrderedDict()
+    queues = []
+    if validation_params is not None:
+        for vtarg in validation_params:
+            vdata_kwargs = copy.deepcopy(validation_params[vtarg]['data'])
```

Comment:
Sure. I think it could be simplified even further (there are many variables that are used once, all named very similarly, so it is hard to follow) but let's deal with that later.